crates/sdk/src/sys.rs

```diff
@@ -1,32 +1,51 @@
 #![allow(dead_code)]
 
 #[cfg(not(target_arch = "wasm32"))]
-mod mock;
 mod types;
 
 pub use types::*;
 
-#[link(wasm_import_module = "env")]
-extern "C" {
-    pub fn panic(loc: Location) -> !;
-    pub fn panic_utf8(msg: Buffer, loc: Location) -> !;
-    // --
-    pub fn register_len(register_id: RegisterId) -> PtrSizedInt;
-    pub fn read_register(register_id: RegisterId, buf: BufferMut) -> Bool;
-    // --
-    pub fn input(register_id: RegisterId);
-    pub fn value_return(value: ValueReturn);
-    pub fn log_utf8(msg: Buffer);
-    pub fn emit(event: Event);
-    // --
-    pub fn storage_read(key: Buffer, register_id: RegisterId) -> Bool;
-    pub fn storage_write(key: Buffer, value: Buffer, register_id: RegisterId) -> Bool;
+macro_rules! wasm_imports {
+    ($module:expr => { $(fn $func_name:ident($($arg:ident: $arg_ty:ty),*) $(-> $returns:ty)?;)* }) => {
+        cfg_if::cfg_if! {
+            if #[cfg(target_arch = "wasm32")] {
+                #[link(wasm_import_module = $module)]
+                extern "C" {
+                    $(
+                        pub fn $func_name($($arg: $arg_ty),*) $(-> $returns)?;
+                    )*
+                }
+            } else {
+                $(
+                    #[no_mangle]
+                    pub extern "C" fn $func_name($($arg: $arg_ty),*) $(-> $returns)? {
```

Comment:
I suspect this \*might\* complain about unused args?

Reply:
Now the only thing that it complains about only-peers build is:

```
warning: methods `post`, `posts`, `create_post`, and `create_comment` are never used
  --> apps/only-peers/src/lib.rs:46:12
   |
45 | impl OnlyPeers {
   | -------------- methods in this implementation
46 |     pub fn post(&self, id: usize) -> Option<&Post> {
   |            ^^^^
...
52 |     pub fn posts(&self) -> &[Post] {
   |            ^^^^^
...
58 |     pub fn create_post(&mut self, title: String, content: String) -> &Post {
   |            ^^^^^^^^^^^
...
84 |     pub fn create_comment(
   |            ^^^^^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: `only-peers` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.69s
```

Reply:
what was the command?

Reply:
`cargo build -p only-peers`

Reply:
Ah, I think that's because the `OnlyPeers` struct is private

Reply:
ah, Wait! Whenever I run `./apps/only-peers/build.sh` I get a lot of `cannot find type Bool in this scope` errors for all of the imported types!

Reply:
How can I use `cargo expand` to expand sys.rs to see the generated code?

Reply:
> How can I use cargo expand to expand sys.rs to see the generated code?

you can use the rust-analyzer option `expand macro recursively at caret`

Reply:
Alternatively, `cargo expand -p calimero-sdk sys`

Reply:
https://github.com/calimero-network/core/pull/417#discussion_r1657416877